### DISCUSSION TITLE
Tech artist teacher tells me that currently it's impossible to make a gouraud shader within Malt

### DISCUSSION THREAD

**Discussion by Schofje** [bnpr/Malt#250] [Q&A] [answered]
Is this true?

Their reasoning is that it results into errors when calling light functions inside a vertex shader context glsl script.

Apparently according to them you cannot do someting like changing the vertex color inthe graph because colors are saved in arrays of 4 vec 4s, that you can only take apart but not put in back together so they fit in the output.

**Comment by pragma37** [comment, marked as answer]
Setting vertex colors from nodes could be improved, but it's doable.

![imagen](https://user-images.githubusercontent.com/36610999/154360160-200aff2b-06f4-4b93-9575-534f8780c63f.png)

[gouraud.zip](https://github.com/bnpr/Malt/files/8083716/gouraud.zip)

Are you using Malt in class?

**Reply by Schofje** [reply]
Oh my god thanks a lot! I'll be dropping a donation in return for making this :)

I am not using MALT in class currently but I am definitely researching different tools for my student film I'll be working on in this semester. I feel like Malt gives me the solutions I need. My tech art teacher is more experienced in OSL and Houdini so this was a fun research for them as well. 
I'll be posting stuff around anytime soon when I show how I use it when it's ready!! :D

**Reply by pragma37** [reply]
Awesome!
If you want to use niche techniques like Gouraud shading, Malt seems like a good candidate.

Thanks for the donation! :)
